### PR TITLE
Updating structures for v0.47.01

### PR DESCRIPTION
**Pull request by quietust**
Updating various globals, one commit at time.

### CODE REVIEW

**Comment by cvuchener** [comment]
I merged 246004f on top of master (2f15182). The material data in items is off by one field:

```
<item_pantsst: 0x7fff88344ac0>
pos                    	 = <coord: 0x7fff88344ac8>
flags                  	 = <item_flags: 0x7fff88344ad0>
flags2                 	 = <item_flags2: 0x7fff88344ad4>
age                    	 = 0
id                     	 = 364
specific_refs          	 = <vector<specific_ref*>[0]: 0x7fff88344ae0>
general_refs           	 = <vector<general_ref*>[2]: 0x7fff88344af8>
world_data_id          	 = -1
world_data_subid       	 = -1
stockpile_countdown    	 = 0
stockpile_delay        	 = 0
unk2                   	 = 0
base_uniform_score     	 = 0
walkable_id            	 = 0
spec_heat              	 = 420
ignite_point           	 = 10508
heatdam_point          	 = 10250
colddam_point          	 = 9900
boiling_point          	 = 60001
melting_point          	 = 60001
fixed_temp             	 = 60001
weight                 	 = 0
weight_fraction        	 = 0
stack_size             	 = 1
history_info           	 = nil
magic                  	 = nil
contaminants           	 = nil
temperature            	 = <temperaturest: 0x7fff88344b58>
wear                   	 = 0
wear_timer             	 = 1
unk_1                  	 = -1
temp_updated_frame     	 = 0
anon_1                 	 = 38
mat_type               	 = 611
mat_index              	 = 572
maker_race             	 = 0
quality                	 = 0
skill_used             	 = -1
maker                  	 = -1
masterpiece_event      	 = 0
improvements           	 = <vector<itemimprovement*>[0]: 0x7fff88344b88>
subtype                	 = <itemdef_pantsst: 0x7fffc1c6dc60>
```

38/611 (naked mode dog leather) should be the type/index for the material. 572 (DWARF) is the maker race.

**Comment by cvuchener** [comment]
I forgot to say that the last report was on linux64. I tested my dwarf therapist memory layout on win64 and there is no issue with materials. I did not recompile dfhack on windows so I cannot compare in details.

File: df.items.xml [hunk 1]
```diff
@@ -694,6 +695,7 @@
 
         <int32_t name='unk_1' init-value='-1' since='v0.34.01'/>
         <int32_t name='temp_updated_frame' init-value='-1'/>
+        <int32_t/>
```
Comment (cvuchener) [inline df.items.xml]: Why was this field added? It seems to be the cause of my issues. I think it only exists as the base class padding with msvc on 64 bits.

Comment (BenLubar) [reply df.items.xml]: Hmm, do we have a way to represent that in df-structures?

Comment (cvuchener) [reply df.items.xml]: It does not need to be represented. It is generated by the compiler.

Comment (BenLubar) [reply df.items.xml]: Actually, looking at your lua output above, it looks like my added field *fixed* the problem you were seeing.

Comment (cvuchener) [reply df.items.xml]: The output was made including this change. `anon_1` is this field, but the value 38 is expected for mat_type.

Comment (cvuchener) [reply df.items.xml]: You did not answer my question. What was this fixing? I cannot simply remove it if it will break something else.

Comment (warmist) [reply df.items.xml]: disclaimer: not an expert
@BenLubar and @cvuchener we have a way to represent it in df-structures and yes it needs to be represented as generated structure is different.E.g:
`
struct A{
int x;
int y;
};
`
and
`
struct A
{
struct {int x;};
int y;
};
`
**might** be laid out differently.

Comment (cvuchener) [reply df.items.xml]: > `struct A { struct {int x;}; int y; };`

This is exactly what this case is **not**. GCC does not add padding at the end of the base class when they are included in a derived class, unlike POD types.

Comment (cvuchener) [reply df.items.xml]: Never mind, I just saw that it was reverted in a6477749abeaddf8c5edcd5174874892b177cce2.

**Pull request merged**